WordPress/gutenberg
# DataViews: Implement `isItemClickable` and `onClickItem` props

## What?


This PR fixes https://github.com/WordPress/gutenberg/issues/66338.
This PR fixes https://github.com/WordPress/gutenberg/issues/66588.

This PR adds two props to the Data Views library:

### onClick

This callback is triggered when the user clicks on a media field or a primary field.

### isClickable

This function determines if a media field or a primary field are clickable. It takes an item as an argument and returns a boolean value indicating whether the item can be clicked.




## Why?


These two callbacks allow developers to implement click handlers for the media field and primary field at the consumer level, rather than within a specific field.


## How?


## Testing Instructions





Ensure that `Quick Edit in DataViews` and `Quick Edit in DataViews` are enabled.

1. Visit the Site Editor
2. Click on pages.
3. Toggle to the table view.
4. Ensure that when you click on the title you are redirected to the page editor
5. Go back.
6. Toggle to the grid view.
7. Ensure that when you click on the image, you are redirected to the page editor.


### Testing Instructions for Keyboard


## Screenshots or screencast
